### DISCUSSION TITLE
SMTP Not Working - OpenSSL Error: wrong version number

### DISCUSSION THREAD

**Discussion by gitelemental** [mastodon/mastodon#19680] [Q&A] [answered]
Hey, I have a self hosted Docker install of Mastodon running through Nginx Proxy Manager on my Synology NAS. Everything seems to working save for emails via PostMark SMTP.

Sidekiq is logging the following errors:

<img width="1173" alt="image" src="https://user-images.githubusercontent.com/114177077/199595570-bdc02251-5761-4bdf-810b-545431dd06dd.png">

Here's my SMTP config:

```
SMTP_SERVER=smtp.postmarkapp.com
SMTP_PORT=587
SMTP_LOGIN=XXX
SMTP_PASSWORD=XXX
SMTP_FROM_ADDRESS=Mastodon <social@sitedomain.com>
SMTP_DOMAIN=sitedomain.com
SMTP_DELIVERY_METHOD=smtp #also failed when set to 'sendmail'
SMTP_AUTH_METHOD=plain
SMTP_OPENSSL_VERIFY_MODE=none
SMTP_ENABLE_STARTTLS_AUTO=false #also failed when set to false
#SMTP_SSL=true #also failed when set to false
#SMTP_TLS=true  #also failed when set to false
```

Any clues behind the Open SSL version error? Could this be related to my proxy setup? 

Thanks in advance.

**Comment by kfitz** [comment]
I have a non-Docker instance that is running into this same error with Sparkpost's SMTP. I've been through a ton of troubleshooting with their support and have discovered that my server can establish a TLS connection with their server, but confirmation messages are still producing the wrong version error in SIdekiq. Any ideas would be super welcome.

**Reply by kfitz** [reply]
I also want to confirm that I have been able to send test messages via SMTP injection to Sparkpost at the command line using the instructions at https://support.sparkpost.com/docs/tech-resources/smtp-injection-with-starttls-using-swaks, so the issue is something in the way Mastodon is configured. I'm at a loss.

**Reply by gitelemental** [reply]
I was able to resolve things on my end with a docker container rebuild and restart. Have you tried restarting the mastodon service after making changes ot the smtp config?

```
sudo systemctl restart mastodon-*
```

https://discourse.joinmastodon.org/t/how-do-you-apply-env-production-changes/1803

**Comment by gitelemental** [comment, marked as answer]
Good news, I was able to get an email to go through successfully. Here's my working SMTP config:

```
SMTP_SERVER=smtp.postmarkapp.com
SMTP_PORT=587
SMTP_LOGIN=xxx
SMTP_PASSWORD=xxx
SMTP_FROM_ADDRESS=name@sitedomain.com
SMTP_OPENSSL_VERIFY_MODE=none
SMTP_DOMAIN=sitedomain.com
```

This time, I followed up the changes with a rebuild and restart of the docker containers with the following commands:

```
docker-compose up -d
docker-compose restart
```

https://github.com/mastodon/mastodon/issues/934#issuecomment-304409856
